cli/src/cmd_auth.rs

```diff
@@ -440,126 +449,97 @@ impl CmdAuthLogout {
-    #[clap(short = 'H', long, env = "OXIDE_HOST", value_parser = parse_host)]
+    /// Specific hostname to validate.
+    #[clap(short = 'H', long, value_parser = parse_host)]
     pub host: Option<url::Url>,
 }
 
 // #[async_trait::async_trait]
 // impl crate::cmd::Command for CmdAuthStatus {
 impl CmdAuthStatus {
     pub async fn run(&self) -> Result<()> {
-        // let cs = ctx.io.color_scheme();
-
         let mut status_info: HashMap<String, Vec<String>> = HashMap::new();
 
-        // let hostnames = ctx.config.hosts()?;
+        let config = Config::default();
+        let mut host_list = config.hosts.hosts;
+
+        // Include login information from environment variables if set
+        if let Some(h) = std::env::var_os("OXIDE_HOST") {
+            let env_token;
+            match std::env::var_os("OXIDE_TOKEN") {
+                Some(t) => env_token = t.into_string().unwrap(),
+                None => env_token = String::from(""),
+            };
+            let info = Host {
+                token: env_token,
+                user: String::from(""),
+                default: false,
+            };
+            host_list.insert(h.into_string().unwrap(), info);
+        }
 
-        // if hostnames.is_empty() {
-        //     writeln!(
-        //         ctx.io.out,
-        //         "You are not logged into any Oxide hosts. Run {} to authenticate.",
-        //         cs.bold("oxide auth login")
-        //     )?;
-        //     return Ok(());
-        // }
+        if host_list.is_empty() {
+            return Err(anyhow!("You are not logged into any Oxide hosts."));
+        }
 
-        let mut failed = false;
-        let mut hostname_found = false;
+        // Return a single result if the --host flag has been set
+        if let Some(url) = &self.host {
+            if host_list.contains_key(url.as_ref()) {
+                host_list.retain(|k, _| k == url.as_str())
+            } else {
+                return Err(anyhow!(
+                    "Host {} Not found in hosts.toml file or environment variables.",
+                    url.as_str()
+                ));
+            }
+        }
 
-        // for hostname in &hostnames {
-        //     if matches!(&self.host, Some(host) if host.as_str() != *hostname) {
-        //         continue;
-        //     }
+        for (host, info) in host_list.iter() {
+            // Construct a client with each host/token combination
+            let client = custom_client(host, info.token.clone());
+
+            let mut host_status: Vec<String> = vec![];
+
+            let result = client.current_user_view().send().await;
+            let user = match result {
+                Ok(usr) => usr,
+                Err(_) => {
+                    host_status.push(String::from(
+                        "Not authenticated. Host/token combination invalid",
+                    ));
+                    status_info.insert(host.to_string(), host_status);
```

Comment:
```suggestion
                    status_info.insert(host.to_string(), "Not authenticated. Host/token combination invalid".to_string());
```

This is what I meant by simpler. Does this work?

Reply:
Ah, no, that doesn't work since `host_status` is a `Vec<String>` and not a `String`. I see why `Vec<String>` feels weird here, but it makes sense for the logged in scenario so its organised neatly like this:

```console
$ oxide auth status 
http://127.0.0.1:12221/: ["Not authenticated. Host/token combination invalid"]
http://127.0.0.1:12220/: ["Logged in to http://127.0.0.1:12220/ as 001de000-05e4-4000-8000-000000004007", "Token: *******************"]

$ oxide auth status -t
http://127.0.0.1:12221/: ["Not authenticated. Host/token combination invalid"]
http://127.0.0.1:12220/: ["Logged in to http://127.0.0.1:12220/ as 001de000-05e4-4000-8000-000000004007", "Token: oxide-token-11111111111111111111111111"]
```

Would you rather we have a very long string?

Reply:
Ok

```suggestion
                    status_info.insert(host.to_string(), vec!["Not authenticated. Host/token combination invalid".to_string()]);
```

This is what I meant by simpler. Does this work?